src/llvm/lib_llvm.cr

```diff
@@ -1,7 +1,28 @@
 require "./enums"
 
-@[Link("stdc++")]
-@[Link(ldflags: "`$(command -v llvm-config-3.6 || command -v llvm-config36 || command -v llvm-config-3.5 || command -v llvm-config35 || command -v llvm-config) --libs --system-libs --ldflags 2> /dev/null`")]
+{% begin %}
+lib LibLLVM
+  LLVM_CONFIG = {{`command -v llvm-config-3.8 || command -v llvm-config38 || command -v llvm-config-3.6 || command -v llvm-config36 || command -v llvm-config-3.5 || command -v llvm-config35 || command -v llvm-config`.chomp.stringify}}
+end
+{% end %}
+
+{% begin %}
+  @[Link("stdc++")]
+  @[Link(ldflags: "`{{LibLLVM::LLVM_CONFIG.id}} --libs --system-libs --ldflags 2> /dev/null`")]
+  lib LibLLVM
+    VERSION = {{`#{LibLLVM::LLVM_CONFIG} --version`.chomp.stringify}}
+  end
+{% end %}
+
+{% begin %}
+  lib LibLLVM
+    IS_38 = {{LibLLVM::VERSION.starts_with?("3.8")}}
+    IS_36 = {{LibLLVM::VERSION.starts_with?("3.6")}}
+    IS_35 = {{LibLLVM::VERSION.starts_with?("3.5")}}
+    IS_36_OR_35 = {{LibLLVM::VERSION.starts_with?("3.6") || LibLLVM::VERSION.starts_with?("3.5")}}
```

Comment:
Well the thought was future proofing, in case we do want to support multiple/more LLVM versions in the future, say somebody has a strong need for 3.7 support, they just add a `IS_37`, or somebody contributes 3.9 support when it's released, they just add a `IS_38`. It's a general pattern easy to understand and expand, instead of trying to figure out and maintain all the necessary combinations here.